src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter.cs

```diff
@@ -111,6 +115,23 @@ internal sealed partial class LocalRewriter : BoundTreeRewriterWithStackGuard
             }
         }
 
+        [Conditional("DEBUG")]
+        private static void AssertValidLocalRewriting(BoundNode node)
+        {
+            switch (node.Kind)
+            {
+                case BoundKind.UsingStatement:
+                case BoundKind.AwaitableValuePlaceholder:
+                    Debug.Assert(false, $"Bound nodes of kind {node.Kind} should not survive past local rewriting");
+                    break;
+            }
+
+            foreach (var child in ((IBoundNodeWithIOperationChildren)node).Children)
```

Comment:
>Children [](start = 75, length = 8)

It doesn't look like we can rely on this API implemented for all nodes. At least a doc comment says: "Override this property to return the child bound nodes **if the IOperation API corresponding to this bound node is not yet designed or implemented**." Also, it looks like it can return nulls in the sequence.